specs/vep-1243-vdk-meta-jobs/README.md

```diff
@@ -206,6 +208,48 @@ This can help to prevent cascading failures and ensure that issues are isolated
 There are more settings that are configurable by the user, that can be explored in the
 [config](/projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_configuration.py).
 
+### Capacity Estimation and Constraints
+
+The load to the Control Service APIs from the Meta Jobs plugin mostly depends on two factors:
+
+* Number of concurrent Data Job Executions (by default limited to 15, but it is
+  [configurable](/projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_configuration.py))
+* Data Job Executions duration
+
+A workflow that consists of one Data Job is going to perform 2 requests per minute to the
+Control Service APIs.
+
+For example, the following workflow consists of 6 Data Jobs:
+
+* `Job 2`, `Job 3` and `Job 4` depend on `Job 1`;
+* `Job 5` depends on `Job 2` and `Job 3`;
+* `Job 6` depends on `Job 4`;
+* The execution time of each of the following jobs is about 1 hour: `Job 1` and `Job 2`;
+* The execution time of each of the following jobs is about 2 hours: `Job 3` and `Job 4`;
+* The execution time of each of the following jobs is about 3 hours: `Job 5` and `Job 6`;
+* The concurrent running jobs limit is configured to 2.
+
+![sample-execution.png](sample-execution.png)
+
+Let's estimate the number of requests the Meta Jobs plugin would send every hour of the workflow execution.
+For our convenience, let's assume that finalizing one job and starting another takes no time (~0 sec).
+There will also be information about the jobs that have run during each time interval.
+
+* During the first hour: about 2 (1 job * 2 reqs/min) requests per minute or 120 requests in total.
+  Running: `Job 1`;
+* During the second hour: about 4 (2 jobs * 2 reqs/min) requests per minute or 240 requests in total.
+  Running jobs: `Job 2` and `Job 3`;
+* During the third hour: about 4 (2 jobs * 2 reqs/min) requests per minute or 240 requests in total.
+  Running jobs: `Job 3` and `Job 4`;
+* During the fourth hour: about 4 (2 jobs * 2 reqs/min) requests per minute or 240 requests in total.
+  Running jobs: `Job 4` and `Job 5`;
+* During the fifth and sixth hour: about 4 (2 jobs * 2 reqs/min) requests per minute or 480 requests in total.
+  Running jobs: `Job 5` and `Job 6`;
+* During the seventh hour: about 2 (1 job * 2 reqs/min) requests per minute or 240 requests in total.
+  Running jobs: `Job 6`.
+
+In total, the vdk-meta-jobs plugin will perform 1560 requests for the whole workflow execution.
```

Comment:
That's a big number just for 6 jobs even for 3 hours. that means that for 6000 jobs we 'd have 1.5 million requests per 3 hours (roughly). 

Let's double check the math and include the META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS into consideration. 
 

Now this assumes job would last hours. 

on average most processing jobs take 10 to 20 minutes with a few outliers taking hours .  
Still it might be worth considering having retry backoff

Reply:
Will recalculate assuming each job takes 20 min and use the default META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS value = 40s
Should I also do the Maths with exponential backoff retry case with factor of 2?

Reply:
Just changed the durations and recalculated the result - updated the text.